UnityProject/Assets/Scripts/Objects/Atmospherics/AcuSample.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using ScriptableObjects.Atmospherics;
+using Systems.Atmospherics;
+
+
+namespace Objects.Atmospherics
+{
+	/// <summary>
+	/// A single sample of the atmosphere for pressure, temperature and gas moles.
+	/// <remarks>See also related classes:<list type="bullet">
+	/// <item><description><seealso cref="AcuSampleAverage"/> represents an atmospheric average of <c>ACU</c> samples</description></item>
+	/// <item><description><seealso cref="AirController"/> Monitors the local atmosphere and controls related devices</description></item>
+	/// <item><description><seealso cref="AcuDevice"/> allows a connection to form between the <c>ACU</c> and devices</description></item>
+	/// </list></remarks>
+	/// </summary>
+	public class AcuSample
+	{
+		/// <summary>In kPa.</summary>
+		public float Pressure = 0;
+		/// <summary>In Kelvin.</summary>
+		public float Temperature = 0;
+		public Dictionary<GasSO, float> GasMoles = new Dictionary<GasSO, float>();
+
+		/// <summary>
+		/// Set the sample values based on the provided <see cref="GasMix"/>.
+		/// </summary>
+		public AcuSample FromGasMix(GasMix mix)
+		{
+			Clear();
+
+			Pressure = mix.Pressure;
+			Temperature = mix.Temperature;
+			foreach (GasValues gas in mix.GasesArray)
+			{
+				GasMoles.Add(gas.GasSO, gas.Moles);
+			}
+
+			return this;
+		}
+
+		public AcuSample Clear()
+		{
+			Pressure = 0;
+			Temperature = 0;
+			GasMoles.Clear();
+
+			return this;
+		}
+	}
+
+	/// <summary>
+	/// A single sample of the atmosphere for pressure, temperature and gas moles.
+	/// <remarks>See also related classes:<list type="bullet">
+	/// <item><description><seealso cref="AcuSample"/> represents a single sample of the atmosphere</description></item>
+	/// <item><description><seealso cref="AirController"/> Monitors the local atmosphere and controls related devices</description></item>
+	/// <item><description><seealso cref="AcuDevice"/> allows a connection to form between the <c>ACU</c> and devices</description></item>
+	/// </list></remarks>
+	/// </summary>
+	public class AcuSampleAverage
```

Comment:
why does this need to be class/thing?
Wouldn't a direct reference to the gas mix or a gas mix class Work?

Reply:
Yep, that's how I originally had it. You can compare the two commits. I decided to create a new class to reduce the complexity in `AirController`. Piggybacking off `GasMix` just felt a little crude, too. Comes with a whole lot of baggage that isn't used.